d2ir/d2ir.go

```diff
@@ -318,11 +318,21 @@ func (f *Field) Copy(newParent Node) Node {
 }
 
 func (f *Field) LastPrimaryRef() Reference {
+	var lastGlobSet Reference
 	for i := len(f.References) - 1; i >= 0; i-- {
-		if f.References[i].Primary() && !f.References[i].DueToLazyGlob() {
-			return f.References[i]
+		if f.References[i].Primary() {
+			if f.References[i].DueToLazyGlob() {
+				if lastGlobSet == nil {
+					lastGlobSet = f.References[i]
+				}
```

Comment:
can this become `return f.References[i]`? when does it need to keep iterating?

Reply:
i thought it was that case, but i realized the last ref always wins, so you're right, it can become that